src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -735,7 +735,6 @@ protected internal void OnDocumentRemoved(DocumentId documentId)
 
         protected virtual void CheckDocumentCanBeRemoved(DocumentId documentId)
         {
-            CheckDocumentIsClosed(documentId);
```

Comment:
why?

Reply:
The commit comment for 774b21cdc2af245cfec8221c5e41cc116af315fd should help. Does that need to be clearer?

The tl;dr though is "it wasn't necessary, already worked anyways, and meant anybody hosting a workspace had to write code to close before removing".

Reply:
Yup.  That sounds reasonable.